graylog2-shared/src/main/java/org/graylog2/shared/security/RestrictToMasterFeature.java

```diff
@@ -0,0 +1,47 @@
+/**
+ * This file is part of Graylog.
+ *
+ * Graylog is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog2.shared.security;
+
+import org.graylog2.plugin.ServerStatus;
+
+import javax.inject.Inject;
+import javax.ws.rs.container.DynamicFeature;
+import javax.ws.rs.container.ResourceInfo;
+import javax.ws.rs.core.FeatureContext;
+import java.lang.reflect.Method;
+
+public class RestrictToMasterFeature implements DynamicFeature {
+    private final ServerStatus serverStatus;
+
+    @Inject
+    public RestrictToMasterFeature(ServerStatus serverStatus) {
+        this.serverStatus = serverStatus;
+    }
+
+    @Override
+    public void configure(ResourceInfo resourceInfo, FeatureContext context) {
```

Comment:
Not completely sure about that, but wouldn't this basically add `RestrictToMasterFilter` to every resource class which contains one or more `@RestrictToMaster` annotations so that no resource method of that resource class could be called on non-master Graylog instances whatsoever?